src/Bicep.Core/Analyzers/Linter/Rules/BCPL1010.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Analyzers.Interfaces;
+using Bicep.Core.Semantics;
+using Bicep.Core.Syntax;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Bicep.Core.Analyzers.Linter.Rules
+{
+    public sealed class BCPL1010 : LinterRule
+    {
+        public BCPL1010() : base(
+            code: "BCPL1010",
+            ruleName: "Parameters must be used",
```

Comment:
Should this be enabled by default? Do we have a similar rule for unused vars?

Reply:
we do have rule for unused vars as well.

Reply:
We decided that all rules would be warnings and enabled by default.